aya/src/maps/mod.rs

```diff
@@ -563,35 +565,37 @@ impl MapData {
         let fd = bpf_get_object(&path_string).map_err(|(_, io_error)| MapError::SyscallError {
             call: "BPF_OBJ_GET",
             io_error,
-        })? as RawFd;
-
-        let info = bpf_map_get_info_by_fd(fd).map_err(|io_error| MapError::SyscallError {
-            call: "BPF_MAP_GET_INFO_BY_FD",
-            io_error,
         })?;
 
+        let info =
+            bpf_map_get_info_by_fd(fd.as_fd()).map_err(|io_error| MapError::SyscallError {
+                call: "BPF_MAP_GET_INFO_BY_FD",
+                io_error,
+            })?;
+
         Ok(MapData {
             obj: parse_map_info(info, PinningType::ByName),
-            fd: Some(fd),
+            fd: Some(fd.into_raw_fd()),
             btf_fd: None,
             pinned: true,
         })
     }
 
-    /// Loads a map from a [`RawFd`].
+    /// Loads a map from an [`OwnedFd`].
     ///
     /// If loading from a BPF Filesystem (bpffs) you should use [`Map::from_pin`](crate::maps::MapData::from_pin).
     /// This API is intended for cases where you have received a valid BPF FD from some other means.
     /// For example, you received an FD over Unix Domain Socket.
-    pub fn from_fd(fd: RawFd) -> Result<MapData, MapError> {
-        let info = bpf_map_get_info_by_fd(fd).map_err(|io_error| MapError::SyscallError {
-            call: "BPF_OBJ_GET",
-            io_error,
-        })?;
+    pub fn from_fd(fd: OwnedFd) -> Result<MapData, MapError> {
```

Comment:
I think we should do this in a separate PR? You're only changing the signature
but otherwise everything else stays the same. Why not do the signature change
when we actually change the impl?

Reply:
That would mean the `bpf_map_info_by_fd` call on line 591 would require an `unsafe` call to [`BorrowedFd::borrow_raw`](https://doc.rust-lang.org/stable/std/os/fd/struct.BorrowedFd.html#method.borrow_raw).

I'm OK with either approach.

Reply:
Yeah the bpf_map_info_by_fd change is also not related to links (it's for maps), so I think it could have gone in a separate PR too (the PR does state this is about links).

`MapData::from_fd` is effectively the only public API change in this PR - without this change I wouldn't have had to look at the PR at all :P

That said no need to add more churn! Thanks a lot @nrxus, looks good I'm approving